transaction-status/src/lib.rs

```diff
@@ -367,22 +384,34 @@ impl UiTransactionStatusMeta {
             fee: meta.fee,
             pre_balances: meta.pre_balances,
             post_balances: meta.post_balances,
-            inner_instructions: meta.inner_instructions.map(|ixs| {
-                ixs.into_iter()
-                    .map(|ix| UiInnerInstructions::parse(ix, &account_keys))
-                    .collect()
-            }),
-            log_messages: meta.log_messages,
-            pre_token_balances: meta
-                .pre_token_balances
-                .map(|balance| balance.into_iter().map(Into::into).collect()),
-            post_token_balances: meta
-                .post_token_balances
-                .map(|balance| balance.into_iter().map(Into::into).collect()),
-            rewards: if show_rewards { meta.rewards } else { None },
-            loaded_addresses: None,
-            return_data: meta.return_data.map(|return_data| return_data.into()),
-            compute_units_consumed: meta.compute_units_consumed,
+            inner_instructions: OptionSerializer::Some(
+                meta.inner_instructions
+                    .map(|ixs| {
+                        ixs.into_iter()
+                            .map(|ix| UiInnerInstructions::parse(ix, &account_keys))
+                            .collect()
+                    })
+                    .unwrap_or_default(),
+            ),
+            log_messages: OptionSerializer::Some(meta.log_messages.unwrap_or_default()),
+            pre_token_balances: OptionSerializer::Some(
+                meta.pre_token_balances
+                    .map(|balance| balance.into_iter().map(Into::into).collect())
+                    .unwrap_or_default(),
+            ),
+            post_token_balances: OptionSerializer::Some(
+                meta.post_token_balances
+                    .map(|balance| balance.into_iter().map(Into::into).collect())
+                    .unwrap_or_default(),
+            ),
+            rewards: OptionSerializer::Some(
+                if show_rewards { meta.rewards } else { None }.unwrap_or_default(),
+            ),
```

Comment:
These fields have new behavior now because now they will never be `null` when the metadata is missing. When metadata is missing, the `OptionSerializer::None` variant should be used.

Reply:
Ach, that's right. These all need to be `into()`s instead of defaults.